### ISSUE TITLE
missing include

### ISSUE THREAD

**Issue by Carmelo908** [marzer/tomlplusplus#232]
## Environment
**toml++ version and/or commit hash:**  
3.4.0

**Compiler:** 
g++ 13.1.0

**C++ standard mode:**  
C++17

**Target arch:**  
x64

**Library configuration overrides:**  
 None

**Relevant compilation flags:**  
None

## Describe the bug
I installed the library with vcpkg and added the toolchain file with CMake, then tried to compile my program and I got an error from the compiler:

.../vcpkg_installed/x64-linux/include/toml++/impl/date_time.hpp:337:17: error: ‘optional’ does not name a type

I went to the file and added "#include <optional>" at the top. it worked and I could compile

**Comment by marzer** [comment]
This should not be the case if you are including the library 'normally'. 🤔 Can you show me the code that caased this, specifically, how were you `#including` toml++?

**Comment by Carmelo908** [comment]
I used #include <toml++/toml.hpp> but I think maybe I'm suposed to just include <optional> in the same file

**Comment by marzer** [comment]
No, you did the right thing. Looks like there's something amiss. I'll look into it, thanks 😃
